source/common/stats/symbol_table_impl.cc

```diff
@@ -173,6 +173,16 @@ void SymbolTableImpl::Encoding::moveToMemBlock(MemBlockBuilder<uint8_t>& mem_blo
   mem_block_.reset(); // Logically transfer ownership, enabling empty assert on destruct.
 }
 
+void SymbolTableImpl::Encoding::appendToMemBlock(StatName stat_name,
+                                                 MemBlockBuilder<uint8_t>& mem_block) {
+  const uint8_t* data = stat_name.dataIncludingSize();
+  if (data == nullptr) {
```

Comment:
what's the use case for appending a stat name with no data? is this just to make it easier in test?

Reply:
no, empty constructor for null-stats and for when there is no tag-extracted-name.

This wasn't necessary before because a StatName constructed from `""` didn't require this special-casing, but I wanted to allow StatName() to also work; without this it crashes.